### ISSUE TITLE
Visual metrics are not recorded (showing 0)

### ISSUE THREAD

**Issue by banuady** [sitespeedio/browsertime#624]
I tried to run SiteSpeed docker container 7.2.2. Here is the command used (Note: I will need to send the actual URL by email):

`docker run --shm-size=1g --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io:7.2.2 url.txt`

For this particular url, I am not getting all the metrics. 

```Google Chrome 67.0.3396.99 
Mozilla Firefox 61.0
[2018-07-23 00:10:08] INFO: Versions OS: linux 4.9.87-linuxkit-aufs nodejs: v8.11.1 sitespeed.io: 7.2.2 browsertime: 3.2.3 coach: 2.0.6
[2018-07-23 00:10:09] INFO: Starting chrome for analysing <URL> 3 time(s)
[2018-07-23 00:10:09] INFO: Testing url <URL> iteration 1
[2018-07-23 00:10:25] INFO: BackEndTime: 2352 DomInteractiveTime: 3626 DomContentLoadedTime: 3720 FirstPaint: 3810 PageLoadTime: 3864
[2018-07-23 00:10:25] INFO: **VisualMetrics FirstVisualChange: 0 SpeedIndex: 0 PerceptualSpeedIndex: 0 LastVisualChange: 0**```

I also noticed the video is not fully recorded, it stopps after 1.13 sec (only 78 requests were recorded).
I tried in Linux (CentOS) and MacOS as well with the same results.

**Comment by soulgalore** [comment]
Hi @banuady thanks for creating the issue! I think this before has been related to testing without any set connectivity, there's some miss-match that things happen too fast. The fastest way to fix that for you is possible following our guides at https://www.sitespeed.io/documentation/sitespeed.io/connectivity/ and try again, hopefully, that will work. But I'm not sure with that firstPaint hmmm, lets see, thanks for sending the URL, I'll have a try asap when I get a chance.

Best
Peter

**Comment by soulgalore** [comment]
Adding it here too (not only in the email): 
One thing you can do to help me is collect the original video and share it with me. Do that by adding:` --browsertime.videoParams.keepOriginalVideo` - that way the original (uncut) video is preserved and I can use that to replicate why Visual Metrics return 0 for all values.

**Comment by banuady** [comment]
Thank you, @soulgalore for this. I did some more testing with throttling as you advise and also collected the original videos. Here are my findings:
- When the throttle is ON, the video is fully recorded and also the visual metrics are captured
- When the throttle is OFF, the video is not recorded properly (although I added --browsertime.videoParams.keepOriginalVideo), so I guess this is why the metrics are not captured.

I downloaded both videos (with throttling ON and OFF): https://www.dropbox.com/sh/i1694wlgsi9d2dh/AACvniPme5Bfytg5KaE8bLyFa?dl=0

In both cases I used the same command (I ran it on my MAC and I used `throttle --up 330 --down 780 --rtt 200`):

`docker run --privileged --shm-size=1g --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io:7.2.3 --browsertime.videoParams.keepOriginalVideo=true url.txt`

**Comment by banuady** [comment]
Hi @soulgalore, I added 1-original.mp4 videos for both tests (throttle on / off).

Regards,
Adrian

**Comment by soulgalore** [comment]
Thanks @banuady ! This was a new thing: It ends the test before the page is fully loaded, I wonder if the pageCompleteCheck need some tweeking. If you have time could you try to enable the other prebuilt alternative: `--browsertime.pageCompleteCheckInactivity` ? There are some more info [here](https://www.sitespeed.io/documentation/sitespeed.io/browsers/#choose-when-to-end-your-test) how we use it.

Best
Peter

**Comment by banuady** [comment]
Thanks @soulgalore ! I did some more tests using `--browsertime.pageCompleteCheckInactivity` and I got all the metrics, even without throttling. (I used the exact script provided with the default 10 sec for loadEventEnd).

Thanks,
Adrian

**Comment by soulgalore** [comment]
Ok, thanks @banuady . I still then can't fully understand why it worked in the first case when you turned on the throttling, I'll look at it when I have the time but will treat it will low prio at the moment since it works now for you.

**Comment by soulgalore** [comment]
We released 4.0 with a total rewrite, please re-open if this is still an issue.
